### DISCUSSION TITLE
MT coverage lower than expected

### DISCUSSION THREAD

**Discussion by rhughwhite** [uclahs-cds/pipeline-call-mtSNV#138] [General]
**Describe the issue**

We've run a set of samples through the pipeline before and after 'top-up' sequencing, with the aim of increased depth of coverage. When looking at the number of reads aligned to chRSRS by MToolBox things look as expected (big increase in aligned reads). However the coverage reported by MToolBox shows a pretty low increase:

**Pre Top-up:**
aligned reads = 307,3800
mean depth = 6,992.724

**Post Top-up:**
aligned reads = 690,9174
mean depth = 7,113.323

So number of aligned reads increases ~2.2X but depth only increases ~ 1.017X. So struggling to make sense of that, anyone have any insight into why this might be?

Coverage files:

> /hot/project/disease/HeadNeckTumor/HNSC-000084-LNMEvolution/pipelines/call-mtSNV/TOP_UP/call-mtSNV-3.0.0/ILHNLNEV000003-T001-P01-F/mitoCaller-1.0.0/intermediate/align-mtDNA-MToolBox_ILHNLNEV000003-T001-P01-F/OUT_extracted_mt_reads_tumour_ILHNLNEV000003-T001-P01-F/extracted-mt-reads-tumour-ILHNLNEV000003-T001-P01-F-coverage.txt

> /hot/project/disease/HeadNeckTumor/HNSC-000084-LNMEvolution/pipelines/call-mtSNV/TOP_UP2/call-mtSNV-3.0.0/ILHNLNEV000003-T001-P01-F/mitoCaller-1.0.0/intermediate/align-mtDNA-MToolBox_ILHNLNEV000003-T001-P01-F/OUT_extracted_mt_reads_tumour_ILHNLNEV000003-T001-P01-F/extracted-mt-reads-tumour-ILHNLNEV000003-T001-P01-F-coverage.txt

Outputs from samtools idxstats*:

> /hot/project/disease/HeadNeckTumor/HNSC-000084-LNMEvolution/pipelines/call-mtSNV/TOP_UP2/call-mtSNV-3.0.0/ILHNLNEV000003-T001-P01-F/mitoCaller-1.0.0/output/samtools_idxstats.txt

> /hot/project/disease/HeadNeckTumor/HNSC-000084-LNMEvolution/pipelines/call-mtSNV/TOP_UP2/call-mtSNV-3.0.0/ILHNLNEV000003-T001-P01-F/mitoCaller-1.0.0/output/samtools_idxstats.txt

*Run on MToolBox BAMs e.g.:
> /hot/project/disease/HeadNeckTumor/HNSC-000084-LNMEvolution/pipelines/call-mtSNV/TOP_UP2/call-mtSNV-3.0.0/ILHNLNEV000003-T001-P01-F/mitoCaller-1.0.0/output/MToolBox-1.2.1-b52269e_HNSC0000016_ILHNLNEV000003-T001-P01-F.bam

**Comment by lydiayliu** [comment]
To add to this, the immediate effect that we have observed is that the "normalized mtDNA copy number" is decreasing as the coverage of the genomic DNA is increasing. To illustrate with an example
- At genomic coverage ~90X, mtDNA coverage was ~150X
- At genomic coverage ~150X, mtDNA coverage was ~90X
- At genomic coverage ~330X, mtDNA coverage is now only ~50X

The formula for normalized copy number I use is
```
copy number = (coverage_mtdna / coverage_genome)(purity * ploidy + (1-purity)*2)
```

**Comment by tnyamaguchi** [comment]
@rhughwhite @lydiayliu One explanation might be that the coverage had been already saturated if the reads were deduplicated. How many base pairs were sequenced for your samples? That is, what is the expected coverage if no deduplication? The MT genome is only ~16.5 kbp.

**Comment by lydiayliu** [comment]
@tyamaguchi-ucla The genomic DNA coverage without deduplication is ~400X

Yeah I think that is a reasonable hypothesis, but it does mean that our formula for calculating "normalized copy number" probably needs to be adjusted?

**Comment by tnyamaguchi** [comment]
> @tyamaguchi-ucla The genomic DNA coverage without deduplication is ~400X

How about the MT DNA coverage without deduplication?

> Yeah I think that is a reasonable hypothesis, but it does mean that our formula for calculating "normalized copy number" probably needs to be adjusted?

Yeah, depending on how we deal with the dedup rate (esp for the MT genome)..

**Comment by lydiayliu** [comment]
> How about the MT DNA coverage without deduplication?

Hmmm I guess that can be calculated from the number of aligned reads? By (read count * read length ) / total genome size

Pre Top-up:
aligned reads (according to MToolBox) = 3,073,800
mean depth (according to MToolBox) = 6,992.724
non-dedup coverage = 3,073,800 * 150*2 / 16,500  = 55,877

Post Top-up:
aligned reads (according to MToolBox) = 6,909,174
mean depth (according to MToolBox) = 7,113.323
non-dedup coverage = 6,909,174 * 150*2 / 16,500  = 125,621

Wow these numbers look a little crazy?

**Comment by tnyamaguchi** [comment]
Assuming MToolBox computes deduped coverage (which we should verify), the genomic region of the MT genome is pretty short, so aligned reads are inevitably duplicated even if they are not actual PRC duplicates, resulting in a significant loss of reads. This is also pretty common in targeted sequencing.

Additionally, MToolBox might apply BQ/MQ filter for coverage calculation.

**Comment by lydiayliu** [comment]
> Assuming MToolBox computes deduped coverage (which we should verify)

Great, thanks!

> Additionally, MToolBox might apply BQ/MQ filter for coverage calculation.

Hmm fair but I can confirm that the top-up sequencing genomic reads pass my BQ/MQ filter as well, so the increase should still be proportional

Anyways everything makes sense but I think it would still be great to get to the bottom of this!

**Comment by tnyamaguchi** [comment]
Sounds good! We might want to convert this issue to GitHub discussion if no objections. @rhughwhite @lydiayliu

**Comment by lydiayliu** [comment]
@tyamaguchi-ucla sure!

> Assuming MToolBox computes deduped coverage (which we should verify)

Will you make this an issue?

**Comment by tnyamaguchi** [comment]
I think @rhughwhite can look into it here and then convert this issue to Discussion once we confirm?

**Comment by rhughwhite** [comment]
There is an optional parameter `UseMarkDuplicates` provided via a config which it looks like is set to false by default - https://github.com/mitoNGS/MToolBox/blob/b52269e98c694d3e4ba25eb80f27b74b48985ddb/MToolBox/MToolBox.sh#L55

This parameter isn't set in the pipeline so I guess it'll be false.
https://github.com/uclahs-cds/pipeline-call-mtSNV/blob/3e5a854ea819174a206e564dff1a47e58b7d7dc2/module/align_mtDNA_MToolBox.nf#L75

That's the only mention I can find of de-duplication in any documentation.

**Reply by lydiayliu** [reply]
Does that mean that the pipeline is NOT removing duplicates by running `picard MarkDuplicates`? Then why is the calculated depth so low?
